src/geometry/algorithms/ArborX_Intersects.hpp

```diff
@@ -412,6 +412,70 @@ struct intersects<PointTag, TetrahedronTag, Point, Tetrahedron>
   }
 };
 
+template <typename Segment>
+struct intersects<SegmentTag, SegmentTag, Segment, Segment>
+{
+  // The algorithm is described in
+  // https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+
+  // Given a segment and a collinear point, check if the point lies on the
+  // segment
+  template <typename Point>
+  KOKKOS_FUNCTION static bool on_segment(Segment const &segment, Point const &p)
+  {
+    using Kokkos::max;
+    using Kokkos::min;
+    auto const &a = segment.a;
+    auto const &b = segment.b;
+    return (p[0] >= min(a[0], b[0]) && p[0] <= max(a[0], b[0])) &&
+           (p[1] >= min(a[1], b[1]) && p[1] <= max(a[1], b[1]));
```

Comment:
Won't compile if the point coordinate type does match with the segment one?
Is that intentional?  If so maybe add a static assert

Reply:
Why won't it compile when point coordinate does not match the segment one? The stuff inside `min` and `max` are coming from the segment, which means same types.

Reply:
FTAD will fail for Kokkos::min/max